### PR TITLE
refactor(ES6): Move forward the codebase to ES6+

### PR DESCRIPTION
**Pull request by nfroidure**
@SebastienElet this is the first part that do not break compatibility but was a first step since the tests were broken on Node10 and the linter was flicking ;). It updates dependencies too.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -95,33 +95,36 @@ FileCache.prototype._readHeader = function _fileCacheReadHeader(data) {
...
 FileCache.prototype.get = function fileCacheGet(key, cb) {
-  var _this = this;
+  const _this = this;
 
-  cb = cb || function() {};
+  cb = cb || (() => {});
 
-  fs.readFile(this._keyToPath(key), function(err, data) {
-    var bucketHeader = null;
+  fs.readFile(this._keyToPath(key), (err, data) => {
```
Comment (SebastienElet) [inline src/index.js:102]: We should `promisify` this part

**Review by SebastienElet — APPROVED** [review]
LGTM 👍 

We may use `util.promisify` to have promise with `fs` module.

**Comment by SebastienElet** [comment]
Ok, for me, can we switch tests to jest ? (i'm working on this and i may do the PR)

**Comment by nfroidure** [comment]
Actually, this was just the most common step between the current module vision and the vision I currently have of its evolution. Here is the next step: https://github.com/SimpliField/sf-filecache/pull/15

It clearly diverges in a lot of ways from the current usage of it so I think you won't merge it and it will probably be the starting point to a fork.

I will add a lock mechanism with the help of external services like https://github.com/nfroidure/common-services#initLockService to avoid race conditions issues and finally add a cleanup function to avoid files to accumulate on the hard drive.

My goal is to make this library compatible with https://github.com/nfroidure/memory-kv-store and its (under construction) Redis based equivalent so that I can substitute one per another depending on my needs / envs.

**Comment by SebastienElet** [comment]
@nfroidure nice work, i will review the #15 asap. 

I merge this one

**Pull request merged**